scipy/signal/_peak_finding.py

```diff
@@ -312,31 +327,38 @@ def peak_prominences(vector, peaks, wlen=None):
     """
     vector = np.asarray(vector)
     peaks = np.asarray(peaks)
+    if peaks.size == 0:
+        # Handle empty peaks
+        if ret_bases:
+            return np.array([]), np.array([]), np.array([])
+        else:
+            return np.array([])
 
-    if vector.ndim != 1:
-        raise ValueError('vector must have exactly one dimension')
+    if vector.ndim != 1 or peaks.ndim != 1:
+        raise ValueError('vector and peaks must have exactly one dimension')
     if peaks.max() >= vector.size:
         raise ValueError('peak position exceeds the range of vector')
     if not np.issubdtype(peaks.dtype, int):
         raise ValueError('peak positions must be integers')
-    if wlen is not None and not 3 <= abs(wlen) <= vector.size:
-        raise ValueError('window lenght must in range [3, vector.size]')
+    if wlen is not None and not 3 <= wlen:
```

Comment:
`if wlen is not None and wlen < 4` and also don't you need to check the integerness of this?

Reply:
Not really. To calculate the window borders I do this:

```python
wleft = peak - int(wlen / 2)
wright = peak + int(wlen / 2)
# Handle border cases
if wleft < 0:
    wleft = 0
elif vector.size < wright:
    wright = vector.size
# Use slice for prominence calculation
window = vector[wleft:wright]
# Correct peak position in vector
peak -= wleft
```
So `wlen / 2` get's floored / converted to an integer anyway.

And 3 is a valid window range although not really useful.

Reply:
Ah  OK, then my first comment is only about using negated ifs. `wlen < 4` is easier to read.

Also in your second code snipped, if the first branch of `if` is done `wleft<0` then it won't go into the second check right? 

you can simply write

    n = vector.size
    window = vector[0 if wleft<0 else wleft:n if n<wright else wright]

one liner ifs are a bit dirty but if you want you can also take them out and write 

    n = vector.size
    wleft = 0 if wleft<0 else wleft
    wright = n if n<wright else wright
    window = vector[wleft:wright]

and so on. I don't mean to be a python grammar-nazi but just suggesting a bit more cleaner flow of code. So that someone else can follow the logic before installing breakpoints and other printing stuff for debugging when needed.

Reply:
I'm happy to learn so feel free to be a Python grammar-nazi. :) I'm happy to accept your second suggestion. But why is `wlen < 4` easier to read than `wlen < 3`?

Reply:
I think that bit is just a typo, his suggestion is essentially the same as mine regarding the `wlen` check -- don't negate with `not`, instead change the relational operator itself without changing the effective truth table